feat: add Redis Sentinel support to Redis lock component documentation (#4771)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-locks/redis-lock.md

| failover              |    N     | Enable failover configuration. Needs sentinelMasterName to be set. The redisHost should be the sentinel host address. See [Redis Sentinel Documentation](https://redis.io/docs/manual/sentinel/). Defaults to `"false"`                                                                              | `"true"`, `"false"`                                             |
| sentinelMasterName    |    N     | The sentinel master name. See [Redis Sentinel Documentation](https://redis.io/docs/manual/sentinel/)                                                                                                                                                                                                              | `"mymaster"`                                                    |

> | sentinelPassword      |    N     | Password for Redis sentinel. No Default. Can be `secretKeyRef` to use a secret reference | `""`, `"KeFg23!"`

Reviewer: ```suggestion
| sentinelPassword      |    N     | Password for Redis Sentinel. No Default. Applicable only when “failover” is true, and Redis Sentinel has authentication enabled | `""`, `"KeFg23!"`
```

{{< /tabpane >}}

> ## Redis Sentinel Behavior

Reviewer: ```suggestion
## Redis Sentinel behavior
```

Use `redisType: "node"` when connecting to Redis Sentinel. Additionally, set `failover` to `"true"` and `sentinelMasterName` to the name of the master node.

> **Failover Characteristics:**

Reviewer: ```suggestion
Failover characteristics:
```

> - **Lock Loss During Failover**: Locks may be lost during master failover if they weren't replicated to the promoted replica before the original master failed

Reviewer: ```suggestion
- Lock loss during failover: Locks may be lost during master failover if they weren't replicated to the promoted replica before the original master failed
```

> - **Failover Window**: Brief unavailability (typically seconds) during automatic master promotion  

Reviewer: ```suggestion
- Failover window: Brief server unavailability (typically seconds) during automatic master promotion  
```

> - **Consistency**: All operations route to the current master, maintaining lock consistency

Reviewer: ```suggestion
- Consistency: All operations route to the current master, maintaining lock consistency
```

> - **Trade-off**: High availability vs. potential lock loss during failover events

Reviewer: ```suggestion
```

> ⚠️ **Important**: Consider your application's tolerance for brief lock loss during failover scenarios.

Reviewer: ```suggestion
{{% alert title="Warning" color="warning" %}}
Consider the trade-off of running Redis with high-availability and failover with the potential of lock loss during failover events. Your application should tolerate brief lock loss during failover scenarios.
{{% /alert %}}
```
